### PR TITLE
Lazyload data in Dataprep table.

### PR DESCRIPTION
**Pull request by ajainarayanan**
#### Approach:
  - First load of dataprep table, load minimal # of rows that gives users data instantly.
  - On scroll load more data.

#### Things added:
  - Added async-to-generator babel plugin to start using `async ... await`. This should be eventually used in lazy loading huge libraries.
  - Removes `babel-preset-es2015` and replaced it with `babel-preset-env` as future versions of babel transpiling will be added to that specific presets.
  - Fixed packages to have specific versions.

#### Things to do:
  - Create a virtual scroll window to recycle dom nodes and render only say 100 nodes at any point in time.
  - Lazy load libraries only used in specific components. First render of entire UI can now be brought down.

### CODE REVIEW

**Comment by bdmogal** [comment]
@ajainarayanan https://issues.cask.co/browse/CDAP-12685

File: cdap-ui/app/cdap/components/DataPrep/DataPrepTable/index.js [hunk 1]
```diff
@@ -31,9 +31,13 @@ import T from 'i18n-react';
...
 import captialize from 'lodash/capitalize';
 import ErrorMessageContainer from 'components/DataPrep/ErrorMessageContainer';
+// Lazy load polyfill in safari as InteresectionObservers are not implemented there yet.
+(async function() {
+  typeof IntersectionObserver === 'undefined' ? await import(/* webpackChunkName: "intersection-observer" */'intersection-observer') : Promise.resolve();
+})();
 
 const PREFIX = 'features.DataPrep.DataPrepTable';
-
+const DEFAULT_WIDOW_SIZE = 100;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepTable/index.js]: nit: `window`, not `widow` right 😛

File: cdap-ui/app/cdap/components/DataPrep/DataPrepTable/index.js [hunk 2]
```diff
@@ -63,8 +68,13 @@ export default class DataPrepTable extends Component {
 
     this.sub = DataPrepStore.subscribe(() => {
       let state = DataPrepStore.getState();
+      if (document.getElementById('dataprep-table-id')) {
+        // Scroll to the top of the table to avoid un-necessary pagination
+        document.getElementById('dataprep-table-id').scrollTop = 0;
+      }
       this.setState({
-        data: state.dataprep.data.map(d => Object.assign({}, d, {uniqueId: shortid.generate()})),
+        windowSize: 100,
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepTable/index.js]: nit: `windowSize: DEFAULT_WIDOW_SIZE`?

**Comment by tonybach** [comment]
The lazy loading works great! I just found this bug though, not sure if it's because of the changes in this PR

data-insights-nan

**Review by elfenheart — APPROVED** [review]
👍  code LGTM

**Comment by leacd** [comment]
LGTM

**Pull request merged**